src/platform/forms-system/src/js/web-component-patterns/fullNamePattern.js

```diff
@@ -219,31 +219,34 @@ const firstNameLastNameSchema = firstNameLastNameDef;
  * schema for `fullNameNoSuffixUI`
  * @returns `commonDefinitions.fullNameNoSuffix`
  */
-const fullNameNoSuffixSchema = {
-  type: 'object',
-  required: ['first', 'last'],
-  properties: {
-    first: {
-      type: 'string',
-      minLength: 1,
-      maxLength: 30,
-    },
-    middle: {
-      type: 'string',
-      maxLength: 30,
-    },
-    last: {
-      type: 'string',
-      minLength: 1,
-      maxLength: 30,
+const fullNameNoSuffixSchema = options => {
+  const { maxLength = 30, minLength = 1 } = options || {};
+  return {
+    type: 'object',
+    required: ['first', 'last'],
+    properties: {
+      first: {
+        type: 'string',
+        minLength,
+        maxLength,
+      },
+      middle: {
```

Comment:
I love what you're doing here, but could you make it so that we could specify the length of first, middle and last independently?  In some of our forms, we want just middle initial, and this would allow us to use this schema.